lib/lrama/state/inadequacy_annotation.rb

```diff
@@ -0,0 +1,128 @@
+# rbs_inline: enabled
+
+module Lrama
+  class State
+    class InadequacyAnnotation
+      # @rbs @state: State
+      # @rbs @token: Symbol
+      # @rbs @actions: Array[Shift | Reduce]
+      # @rbs @contribution_matrix: Hash[Shift | Reduce, Hash[Item, bool]]
+      attr_accessor :state, :token, :actions, :contribution_matrix
+
+      # @rbs (State state, Symbol token, Array[Shift | Reduce] actions, Hash[Shift | Reduce, Hash[States::Item, bool]] contribution_matrix) -> void
+      def initialize(state, token, actions, contribution_matrix)
```

Comment:
Note:  Inadequacy Annotation is defined as a tuple of (n_i, γ) in Definition 3.28. n_i is an inadequacy manifestation description which is a tuple of (state, token, actions). Then instance of this class has necessary and sufficient information.